pkg/features/features.go

```diff
@@ -88,8 +103,24 @@ func NewFeatureGate() featureFlag {
 	return fg
 }
 
+// checkRequired check the feture requirement of other requirement is ok
+func (fg featureFlag) checkRequired(f Feature) bool {
+	if requireds, ok := featureRequirements[f]; ok {
+		for f, v := range requireds {
+			if fg[f] != v {
+				return false
+			}
+		}
+	}
+	return true
+}
+
 // IsEnabled returns true if the feature is enabled
 func (fg featureFlag) IsEnabled(f Feature) bool {
+	if !fg.checkRequired(f) {
```

Comment:
This need not be checked everytime, since we set the flag to default value initially.